drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -421,11 +421,15 @@ static void start_ble_rf(void)
 	LL_RCC_SetRTCClockSource(LL_RCC_RTC_CLKSOURCE_LSI);
 #endif
 
-	/* Set RNG on HSI48 */
+#if !STM32_HSI48_ENABLED
+#warning Bluetooth IPM requires HSI48 clock to be enabled
+	/* Deprecated: Set BLE on HSI48 */
 	LL_RCC_HSI48_Enable();
 	while (!LL_RCC_HSI48_IsReady()) {
 	}
 
+#endif /* !STM32_HSI48_ENABLED */
+	/* HSI48 Clock is enabled through the DTS */
 	LL_RCC_SetCLK48ClockSource(LL_RCC_CLK48_CLKSOURCE_HSI48);
```

Comment:
Defining CLK48 clock source should actually be done using clock_control driver using a [clock mux](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/clock_control/clock_stm32_mux.c).

See [h7 example](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/clock_control/stm32_clock_configuration/stm32h7_devices/boards/spi1_per_ck_hsi.overlay).

So:
- this call should be removed.
- In theory, we should define a clk_48 clock mux node that selects HSI48 as source (could be default in stm32wb.dtsi). Though since here we only default value, this would be adding a (simple) driver for nothing. So this could be left as disabled in stm32wb.dtsi file
- on nucleo_wb, add the node with `clocks = <&rcc STM32_SRC_HSI48 CLK48_SEL(0)>;` as example. But keep it disabled to save some footprint (since this is the default value)